inbm/dispatcher-agent/dispatcher/sota/sota.py

```diff
@@ -356,7 +357,7 @@ def execute_from_manifest(self,
                     self._update_logger.status = OTA_PENDING
                     self._update_logger.error = ""
                 self._update_logger.save_log()
-                if self.sota_mode == 'download-only' or self._device_reboot in ["No", "N", "n", "no", "NO"]:  # pragma: no cover
+                if self.sota_mode == 'download-only' or rebooter.is_reboot(self._device_reboot):
                     self._dispatcher_broker.telemetry("No reboot (SOTA pass)")
                 else:
                     self._dispatcher_broker.telemetry("Going to reboot (SOTA pass)")
```

Comment:
**Suggestion:** Refactor the conditional expression to improve readability and maintainability. [maintainability]
**NOTE:** Ensure that the suggested code is comprehensive. This serves as an example of the code you could use. The user should verify it after accepting the suggestion.
```suggestion
                reboot_message = "No reboot (SOTA pass)" if self.sota_mode == 'download-only' or rebooter.is_reboot(self._device_reboot) else "Going to reboot (SOTA pass)"
                self._dispatcher_broker.telemetry(reboot_message)
```